### PR TITLE
Do not panic when a Pod is not Waiting

### PR DESCRIPTION
**Pull request by qbarrand**
This PR addresses BZ 2039238.
The bug was introduced [here](https://github.com/openshift-psap/special-resource-operator/pull/205/files#diff-a7ff0d3ae4aee6ef9557be809cb7b567203efb928f66ef45fee67edbf8dedf3cL322-L329). [This test case](https://github.com/openshift-psap/special-resource-operator/pull/230/files#diff-ad2d965528c17752e089494fa689b6cc4b469b65b11da80ad8f40264fc806f14R651) addresses it.
The PR temporarily makes `CheckForImagePullBackOff` a member of the `Creator` interface so that we can unit test it. In the long run, many methods (including `CheckForImagePullBackOff`) should be extracted to a new internal type. See [MGMT-8567](https://issues.redhat.com/browse/MGMT-8567).

### CODE REVIEW

File: pkg/resource/resource.go [hunk 1]
```diff
@@ -38,6 +39,8 @@ var (
 //go:generate mockgen -source=resource.go -package=resource -destination=mock_resource_api.go
 
 type Creator interface {
+	// TODO(qbarrand) remove CheckForImagePullBackOff from the interface and extract to an internal type
+	CheckForImagePullBackOff(context.Context, *unstructured.Unstructured, string) error
```
Comment (yevgeny-shnaidman) [inline pkg/resource/resource.go]: why do we need to add this function to the interface? it is only used internally, no?

Comment (qbarrand) [reply pkg/resource/resource.go]: This is so we can unit test it, as the tests currently belong to `resources_test`; adding this function to the interface was a smaller change than moving the tests to `resources`. In the long run, I want to move most functions to an internal package.

Comment (yevgeny-shnaidman) [reply pkg/resource/resource.go]: we already have test that are in the same package as the unit they are testing. I think that changing production code to support the unit-test  is worse then changing the unit-test to support production

Comment (qbarrand) [reply pkg/resource/resource.go]: Moved the tests to the `resource` package. @yevgeny-shnaidman PTAL 🙂

**Pull request merged**